src/QuadForm/ClosestVector.jl

```diff
@@ -69,20 +59,26 @@ function pojective_quadratic_triple(Q::MatrixElem, L::MatrixElem, c::RingElement
         push!(pp1, p1)
         push!(LL1, L1)
         push!(r1, Q[n+1-i, n+1-i])
-        push!(m1, L[n+1-i,1:1])
+        push!(m1, L[n+1-i,1:1][1])
         pL =[Lqqt, pp1, LL1, r1, m1]
     end
     #------------------------------------------------
+    c1 = c; pQT = []; ppQT = []; pLL = []; 
+    ppLL = []; cc1 = []; Lqt = [];
     for j in 1:size(pL[1],1)
         pt1 = transpose(pL[2][j])
         pQT = pL[1][j] - pL[4][j]^-1 * (pL[2][j] * pt1)
-        pLL = pL[3][j] - ((pL[5][j] * pL[4][j]^-1)[1] * pL[2][j])
-        c1 = c1 - (pL[5][j]^2 * pL[4][j]^-1)[1] #rational number
+        pLL = matrix(pL[3][j]) - ((pL[5][j] * pL[4][j]^-1) * pL[2][j])
+        PLL = typeof(L)(undef, size(pLL)[1])
+        for j2 in 1:size(pLL)[1]
+            PLL[j2] = pLL[j2]
+        end
```

Comment:
This looks like you are copying `pLL` and give the copy the name`PLL`. Why? Before you did not copy.

Reply:
I do that to convert the type of the list from ```fmpq_mat``` to ```Vector{fmpq}``` So that the output tuple type is consistent with the input parameters. I can leave the result as ```fmpq_mat```. But I can also change the input parameter ```Vector{fmpq}``` to a 1xn ```fmpq_mat```. Then the output would also be consistent with the input types. 
What would you recommend?

Reply:
I would recommend to think this carefully through: A `Vector{fmpq}` is a julia-construct that we interpret as either row or column depending on context. However, the memory layout is different. For fmpq it's not too bad, but for fmpz the performance on say addition of Vector{fmpz} and fmpz_mat could easily be a magnitude, due to different memory layout...
For more interesting operations this can get more and more interesting. 
What do you want to do?

Reply:
Thank you for your comment. That is good to know. I will keep this in consideration for future code. For the moment, this file constitutes of pre-procedures required for the enumeration of closest vectors. Most of the functions are not optimized and will change over time. In practice, the user should only call one of the ```closest_vectors()``` functions at the end of this file and the rest would eventually be private functions (hidden away unless someone needs them). The one that takes ZLat (as an input parameter) is the optimized one at the moment.

Reply:
Do you actually need rational matrices, i.e. are you performing a divison anywhere?
Then `fmpz` is the better choice or perhaps even `Int` if you can bound the ellipsoid and any other integers appearing.